UnityProject/Assets/Scripts/Shuttles/ShuttleInteract.cs

```diff
@@ -0,0 +1,30 @@
+﻿using PlayGroups.Input;
+using UI;
+using UnityEngine;
+
+public class ShuttleInteract : InputTrigger 
+{
+    public string interactionMessage;
+    public MatrixMove ShuttleMatrixMove;
+    private GameObject player;
+    
+    // Method that brings up shuttle controls.
+    public override void Interact(GameObject originator, Vector3 position, string hand)
+    {
+        player = originator;
+        ControlTabs.ShowShuttleTab(ShuttleMatrixMove);
```

Comment:
In this case tab opening should be granted by server.
In fact, server should know when it's opened on clients and when it's not, it would be required for pushing radar updates etc.
So [Command] should be introduced here, or InteractMessage should be sent.
(At this point it's not very intuitive, I know)
So, InteractMessage way would look like this:
```cs
	public override void Interact(GameObject originator, Vector3 position, string hand)
		{
                      var playerScript = originator.GetComponent<PlayerScript>();
                      if (playerScript.canNotInteract() || !playerScript.IsInReach( gameObject.transform.position ))
                      { //check for both client and server
                          return;
                      }
			if (!isServer)
			{ 
			        //Client wants this code to be run on server
				InteractMessage.Send(gameObject, hand);
			}
			else
			{
				//Server actions
				//you'll have to write this yourself. it would call Open() for client when Process()'ed 
                               TabActionMessage.Send( originator, console, TabAction.Open ); 
                               //... some code to mark this one as opened by someone  
			}
		}
```

And Command way would be like that:
```cs
  // Method that brings up shuttle controls.
    public override void Interact(GameObject originator, Vector3 position, string hand)
    {
        player = originator;
        CmdShowShuttleTab( gameObject, originator );
    }

    /// Request to open shuttle tab
    [Command]
    public void CmdShowShuttleTab( GameObject console, GameObject originator ) {
        var playerScript = originator.GetComponent<PlayerScript>();
        if (playerScript.canNotInteract() || !playerScript.IsInReach( gameObject.transform.position ))
        { //check for server
            return;
        }
        //you'll have to write this yourself. it would call Open() for client when Process()'ed 
        TabActionMessage.Send( originator, console, TabAction.Open ); 
        //... some code to mark this one as opened by someone  
    }

    public void Open() {
        ControlTabs.ShowShuttleTab(ShuttleMatrixMove);
    }

    public void Close() {
        ControlTabs.HideShuttleTab();
    }
....
public interface ITab
{
    void Open();
    void Close();
}
```